### PR TITLE
CAS-1083 Docs - Restructure ChatUI docs

### PR DESCRIPTION
**Pull request by samiuelson**
https://stream-io.atlassian.net/browse/CAS-1083

### 🎯 Goal

* Restructure ChatUI docs
* Add missing kdocs to ChatUI

### 🧪 Testing

* Run docs locally `npx stream-chat-docusaurus -i -s`
* Open ChatUI section and read it.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### CODE REVIEW

File: docusaurus/docs/Android/03-ui/03-chatui.md [hunk 1]
```diff
@@ -1,54 +1,118 @@
...
+ * Avatar view bitmap transformation
+ * Attachments urls overriding
+
+The full list of `ChatUI` properties you can override include:
+ * `ChatUI.supportedReactions` Allows overriding default set of supported Message reaction.
+ * `ChatUI.mimeTypeIconProvider` Allows defining own icons for different mime types.
+ * `ChatUI.fonts` Allows specifying default font for TextViews displayed by UI components. 
+ * `ChatUI.avatarBitmapFactory` Allows intercepting the avatar Bitmap and modifying it before it's displayed by the `AvatarView`.
+ * `ChatUI.urlSigner` Allows adding authorization tokens for images, video, etc.
+ * `ChatUI.markdown` Interface to to customize the markdown parsing behaviour, useful if you want to provide your custom markdown parsing logic, use more markdown modules, etc.
```
Comment (leandroBorgesFerreira) [inline docusaurus/docs/Android/03-ui/03-chatui.md:20]:
```suggestion
 * `ChatUI.markdown` Interface to customize the markdown parsing behaviour, useful if you want to provide your custom markdown parsing logic, use more markdown modules, etc.
```

File: docusaurus/docs/Android/03-ui/03-chatui.md [hunk 2]
```diff
@@ -1,54 +1,118 @@
...
+|![Light_mode](../assets/chat_ui_custom_reaction.png)|![Dark_mode](../assets/chat_ui_custom_reaction_active.png)|
 
-To display some screens available in the SDK, it is necessary to navigate to them.
-The SDK does that using the ChatNavigator and it navigates to:
-- `AttachmentGaleryActivity`: To display the gallery of pictures
-- After a click in a link, to the destination of the link (the user leaves the chat).
+### Custom MIME type icons
 
-It is possible to add a custom navigator to `ChatUI` by adding a new `ChatNavigator`.
+When possible SDK displays thumbnails for image files. Mime type icons are displayed for other files at `MessageListView` as attachments icons and at files gallery screen.
```
Comment (leandroBorgesFerreira) [inline docusaurus/docs/Android/03-ui/03-chatui.md:55]:
```suggestion
When possible, SDK displays thumbnails for image files. Mime type icons are displayed for other files at `MessageListView` as attachments icons and at files gallery screen. 
```

File: docusaurus/docs/Android/03-ui/03-chatui.md [hunk 3]
```diff
@@ -1,54 +1,118 @@
...
 
-To display some screens available in the SDK, it is necessary to navigate to them.
-The SDK does that using the ChatNavigator and it navigates to:
-- `AttachmentGaleryActivity`: To display the gallery of pictures
-- After a click in a link, to the destination of the link (the user leaves the chat).
+### Custom MIME type icons
 
-It is possible to add a custom navigator to `ChatUI` by adding a new `ChatNavigator`.
+When possible SDK displays thumbnails for image files. Mime type icons are displayed for other files at `MessageListView` as attachments icons and at files gallery screen. 
+By default SDK provides built-in MIME type icons for the most popular file types, and displays a generic file icon for others.
```
Comment (leandroBorgesFerreira) [inline docusaurus/docs/Android/03-ui/03-chatui.md:56]:
```suggestion
By default SDK provides built-in MIME type icons for the most popular file types and displays a generic file icon for others.
```

File: docusaurus/docs/Android/03-ui/03-chatui.md [hunk 4]
```diff
@@ -1,54 +1,118 @@
...
+        else -> R.drawable.stream_ui_ic_file
+    }
 }
 ```
 
-Following there's a simple example of a customization of destination:
+### Transforming avatar bitmap with AvatarBitmapFactory
+
+It is possible to override `AvatarBitmapFactory` so the avatars will
+be generated accordingly to the new configuration. It is possible to configure
```
Comment (leandroBorgesFerreira) [inline docusaurus/docs/Android/03-ui/03-chatui.md:82]: `You can configure the user`... You we don't write `It is possible` twice

Comment (samiuelson) [reply docusaurus/docs/Android/03-ui/03-chatui.md]: Fixed

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**